libs/ledger/src/chain/block_coordinator.cpp

```diff
@@ -1090,64 +1170,63 @@ BlockCoordinator::State BlockCoordinator::OnReset()
 
   FETCH_LOG_INFO(LOGGING_NAME, "Reset condition");
 
-  // trigger stake updates at the end of the block lifecycle
-  if (stake_ && beacon_)
+  Block const *block = nullptr;
+
+  if (next_block_)
   {
-    Block const *block = nullptr;
+    block = next_block_.get();
+  }
+  else if (current_block_)
+  {
+    block = current_block_.get();
+  }
+  else
+  {
+    FETCH_LOG_ERROR(LOGGING_NAME, "Unable to find a previously executed block!");
+  }
 
-    if (next_block_)
-    {
-      block = next_block_.get();
-    }
-    else if (current_block_)
-    {
-      block = current_block_.get();
-    }
-    else
-    {
-      FETCH_LOG_ERROR(LOGGING_NAME,
-                      "Unable to find a previously executed block when updating staking!");
-    }
+  // trigger stake updates at the end of the block lifecycle
+  if (stake_ && block)
+  {
+    stake_->UpdateCurrentBlock(*block);
+  }
 
+  if (beacon_)
+  {
     // If we are on the head of the main chain, and it is a 'committee triggering' block
-    if (block && (*chain_.GetHeaviestBlock() == *block))
+    // and we are in sync.
+    // TODO(HUT): this needs very carefully looking at!
```

Comment:
Issue number

Reply:
Is this a comment for us to review carefully, and what is your concern?

Reply:
It was about the mechanics of committee renewal and forking - I will delete the comment since I think I can address it in the next PR